File: Codeforces Submissions/CF 799/F.cpp
#include <iostream>
#include <math.h>
using namespace std;
	int main()
	{
		int t ;
		cin >> t;
		bool sol[t];
		for (int i = 0; i<t; ++i)
		{
			sol[i] = false;
			int n ; cin >> n;
			int a[n]; int b[10];
			for (int j = 0; j<n ; ++j)
			{
				cin >> a[j];
				b[a[j]%10]++;
			}
			int nn = 10;
			for (int j = 0; j<nn && !sol[i] ; ++j)
			{
				for (int k = 0; k<nn && !sol[i] ; ++k)
				{
					for (int l = 0; l<nn && !sol[i] ; ++l)
					{
						for (int jj = 0; jj<=b[j] && jj<4; ++jj)
						{
							for (int kk = 3-jj; kk<=b[k] && kk<4; ++kk)
							{
								for (int ll = 3-jj-kk; ll<=b[l] && ll<4; ++ll)
								{
									if ((j*jj + k*kk + l*ll)%10 == 3)
										sol[i] = true;
								}
							}
						}
					}
				}
			}
		}
		for (int i = 0; i<t; ++i)
		{
			if (sol[i])
				cout << "YES" <<"\n";
			else 
				cout << "NO" <<"\n";
		}
		return 0;
	}

File: Codeforces Submissions/CF 799/F.java
import java.util.*;
public class F
{
	public static void main(String args[])
	{
		Scanner sc = new Scanner(System.in);
		int t = Integer.parseInt(sc.nextLine());
		boolean sol[] = new boolean[t];
		for (int i = 0; i<t; ++i)
		{
			sol[i] = false;
			int n = Integer.parseInt(sc.next().trim());
			int a[] = new int[n];
			for (int j = 0; j<n ; ++j)
				a[j] = Integer.parseInt(sc.next());
			for (int j = 0; j<n && !sol[i] ; ++j)
			{
				for (int k = j+1; k<n && !sol[i] ; ++k)
				{
					for (int l = k+1; l<n && !sol[i] ; ++l)
					{
						if ((a[j]+a[k]+a[l])%10 == 3)
							sol[i] = true;
					}
				}
			}
		}
		for (int i = 0; i<t; ++i)
		{
			if (sol[i])
				System.out.println("YES");
			else 
				System.out.println("NO");
		}
	}
}